### ISSUE TITLE
Do not fall back to pattern match for rewrite as rewrite is not a syntactic sugar anymore

### ISSUE THREAD

**Issue by ghost** [agda/agda#2087]
Since rewrite is no longer a syntactic sugar, agda should not transform rewrites  into pattern matching when doing case splitting
After case splitting, rewrite is transformed into pattern matching.
I am using Agda 2.5.1

**Comment by andreasabel** [comment]
Duplicate of #1735.

**Comment by UlfNorell** [comment]
The problem here is that we don't have syntax for "case-split after rewrite". Patterns and rewrites are all in a single clause and the type checker interprets that as "split first, then rewrite". The naive fix to this issue, i.e. keeping the rewrite, would generate bad left-hand sides in the cases where the rewrite really needs to happen first. Translating (albeit imperfectly) to a `with` can make those examples work.

**Comment by jespercockx** [comment]
Perhaps we should just refuse to split when there is a `rewrite`?
